### PR TITLE
chore: updated turborepo config

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR updates the `turborepo` configuration updating the matching globs for each command. This should also fix the issue of Turborepo accidentally marking a specific file as unchanged.

This PR also updates some `env` variables and the target for the `pull-request.yaml` workflow

### CODE REVIEW

**Comment by ovflowd** [comment]
cc @nodejs/website if we can fast-track this!

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -45,16 +48,10 @@ jobs:
...
-          TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
-          TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
+        run: npx turbo lint --filter=$TURBO_REF_FILTER
 
       - name: Run Prettier
-        run: npx turbo prettier
-        env:
-          TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
-          TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
+        run: npx turbo prettier --filter=$TURBO_REF_FILTER
```
Comment (bmuenzenmeyer) [inline .github/workflows/pull-request.yml:54]: will this exhibit the same Windows problem?

I know https://github.com/nodejs/nodejs.org/actions/runs/5575298609/jobs/10185031667 is green but it seems odd

Comment (ovflowd) [reply .github/workflows/pull-request.yml:54]: No? This step runs only on Ubuntu

**Comment by ovflowd** [comment]
cc @nodejs/website if we can fast-track this!

**Review by bmuenzenmeyer — APPROVED** [review]
good work on this. cross platform dev is fun aint it?

**Comment by ovflowd** [comment]
> good work on this. cross platform dev is fun aint it?

Let's not even talk about that lol

**Pull request merged**